### ISSUE TITLE
Results from SSD is wrong after serialized and deserialized.

### ISSUE THREAD

**Issue by pagepage800825** [NVIDIA/TensorRT#755]
## Description

I have followed the instruction in samplels/opensource/sampleSSD
The result was good, and shows
Label: car, confidence: 96.0587 xmin: 4.14486 ymin: 117.443 xmax: 244.102 ymax: 241.829

And then, I would like to serialize the engine to file for faster building again.
I do the following in the end of SampleSSD::infer:
IHostMemory* trtModelStream;
    trtModelStream = mEngine->serialize();
    if (trtModelStream == nullptr)
    {
        std::cout << "Engine serialization failed" << std::endl;
        return false;
    }

and then try to deserialize it to engine again:

    Logger logger;
    IRuntime* runtime{nvinfer1::createInferRuntime(logger)};
    ICudaEngine* engine2 = runtime->deserializeCudaEngine(trtModelStream->data(), trtModelStream->size(), nullptr);
    std::shared_ptr<nvinfer1::ICudaEngine> mEngine2(engine2, samplesCommon::InferDeleter());

finally do inference using mEngine2 (which is simply serialized and then deserialize from mEngine

    samplesCommon::BufferManager buffers2(mEngine2, mParams.batchSize);

    auto context2 = SampleUniquePtr<nvinfer1::IExecutionContext>(mEngine2->createExecutionContext());
    if (!context2)
    {
        return false;
    }

    // Read the input data into the managed buffers
    assert(mParams.inputTensorNames.size() == 1);
    if (!processInput(buffers2))
    {
        return false;
    }

    // Memcpy from host input buffers to device input buffers
    buffers2.copyInputToDevice();

    bool status2 = context2->execute(mParams.batchSize, buffers2.getDeviceBindings().data());
    if (!status2)
    {
        return false;
    }

    // Memcpy from device output buffers to host output buffers
    buffers2.copyOutputToHost();

    // Post-process detections and verify results
    if (!verifyOutput(buffers2))
    {
        return false;
    }

In fact, this is just a copy-paste from SampleSSD::infer()
While the result is weird:

## Environment

**TensorRT Version**: 
**GPU Type**: 
**Nvidia Driver Version**: 
**CUDA Version**: 
**CUDNN Version**: 
**Operating System + Version**: 
**Python Version (if applicable)**: 
**TensorFlow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if container which image + tag)**:

**Comment by HappyKerry** [comment]
> ## Description
> After serialize and deserialize the tensorrt engine for SSD, the result changes
> 
> ## Environment
> **TensorRT Version**: 7.1.3
> **GPU Type**: Jetson nano
> **Nvidia Driver Version**:
> **CUDA Version**: 10.2
> **CUDNN Version**:
> **Operating System + Version**: Jetson nano JetPack 4.4 (Ubuntu 18.04.4)
> **Python Version (if applicable)**:
> **TensorFlow Version (if applicable)**:
> **PyTorch Version (if applicable)**:
> **Baremetal or Container (if container which image + tag)**:
> 
> ## Relevant Files
> https://github.com/NVIDIA/TensorRT/tree/master/samples/opensource/sampleSSD
> 
> ## Steps To Reproduce
> I have followed the instruction in samplels/opensource/sampleSSD
> The result was good, and shows
> 
> ```
> Label: car, confidence: 96.0587 xmin: 4.14486 ymin: 117.443 xmax: 244.102 ymax: 241.829
> ```
> 
> Which is the expected result.
> 
> And then, I would like to serialize the engine to file for faster building again.
> I do the following in the end of SampleSSD::infer:
> 
> ```
> IHostMemory* trtModelStream;
> trtModelStream = mEngine->serialize();
> if (trtModelStream == nullptr)
> {
>     std::cout << "Engine serialization failed" << std::endl;
>     return false;
> }
> ```
> 
> and then try to deserialize it back to engine:
> 
> ```
> Logger logger;
> IRuntime* runtime{nvinfer1::createInferRuntime(logger)};
> ICudaEngine* engine2 = runtime->deserializeCudaEngine(trtModelStream->data(), trtModelStream->size(), nullptr);
> std::shared_ptr<nvinfer1::ICudaEngine> mEngine2(engine2, samplesCommon::InferDeleter());
> ```
> 
> finally do inference using mEngine2 (which is simply serialized and then deserialize from mEngine
> The following code is just a copy-paste from SampleSSD::infer(), except changing variable names.
> 
> ```
> samplesCommon::BufferManager buffers2(mEngine2, mParams.batchSize);
> 
> auto context2 = SampleUniquePtr<nvinfer1::IExecutionContext>(mEngine2->createExecutionContext());
> if (!context2)
> {
>     return false;
> }
> 
> // Read the input data into the managed buffers
> assert(mParams.inputTensorNames.size() == 1);
> if (!processInput(buffers2))
> {
>     return false;
> }
> 
> // Memcpy from host input buffers to device input buffers
> buffers2.copyInputToDevice();
> 
> bool status2 = context2->execute(mParams.batchSize, buffers2.getDeviceBindings().data());
> if (!status2)
> {
>     return false;
> }
> 
> // Memcpy from device output buffers to host output buffers
> buffers2.copyOutputToHost();
> 
> // Post-process detections and verify results
> if (!verifyOutput(buffers2))
> {
>     return false;
> }
> ```
> 
> While the result is weird:
> 
> ```
> Label: car, confidence: 96.0587 xmin: 79 ymin: 143 xmax: 241 ymax: 300
> Label: car, confidence: 86.8168 xmin: 109.449 ymin: 166.724 xmax: 300 ymax: 281.276
> Label: car, confidence: 84.5954 xmin: 38.7244 ymin: 109.449 xmax: 153.276 ymax: 300
> Label: car, confidence: 77.4324 xmin: 0 ymin: 166.724 xmax: 210.551 ymax: 281.276
> Label: car, confidence: 70.3909 xmin: 207 ymin: 143 xmax: 300 ymax: 300
> Label: car, confidence: 68.9535 xmin: 0 ymin: 204.951 xmax: 115.048 ymax: 300
> Label: car, confidence: 65.7628 xmin: 264.756 ymin: 161.511 xmax: 300 ymax: 300
> ```
> 
> The top-1 score is the same, but the coordinate changes.
> The methods I serialize the model works in Tensorrt 5, but does not worked in Tensorrt7.
> I would guess there are something bugs in serialization of SSD-related plugins?
> I noticed that some coordinates are integer, I would guess that the output is prior-box coordinate after deserialization, but not location, while I could not check.

hello，have you solved the problem？I moved the sampleSSD code ，and compiled in tensorrt7.1.3，got the wrong detections like yours

**Comment by coolbei** [comment]
with tensorRT 7.0, it just exit on assert:

**Comment by ttyio** [comment]
Hello @pagepage800825 , please check https://github.com/NVIDIA/TensorRT/issues/1089,
there is bug in the plugin serialization, and need rebuild OSS plugin, thanks
